be1-go/internal/popserver/database/database.go

```diff
@@ -0,0 +1,76 @@
+package database
+
+import (
+	"golang.org/x/xerrors"
+	"popstellar/internal/popserver/database/repository"
+	"popstellar/message/answer"
+	"sync"
+	"testing"
+)
+
+var once sync.Once
+var instance repository.Repository
+
+func InitDatabase(db repository.Repository) bool {
+	hasInit := false
+
+	once.Do(func() {
+		instance = db
+		hasInit = true
+	})
+
+	return hasInit
+}
+
+func SetDatabase(t *testing.T) (*repository.MockRepository, error) {
```

Comment:
Same as for `SetConfig()` but here you could have `SetDatabase(Repo)` and create the mock repository in your test. It would allow much better separation between testing and production code.